manifests/init.pp

```diff
@@ -40,20 +40,14 @@
 #  The sys_id of the user assigned to the incident as specified in the
 #  sys_user table. Note that if assignment_group is also specified, then
 #  this must correspond to a user who is a member of the assignment_group.
-# @params [Array[Enum['failed_changes', 'corrective_changes', 'intentional_changes', 'pending_changes', 'no_changes'], 1, 3]] incident_creation_report_statuses
-#   The kinds of report attributes that can trigger an incident to be sent to
-#   Servicenow. Choose any of ['failed_changes', 'corrective_changes',
-#   'intentional_changes', 'pending_changes', 'no_changes']. The report
-#   processor will create incidents for reports that have at least one of the
-#   specified attributes. So if you use e.g. the default value
-#   (['failed_changes','corrective_changes']), then the report processor will
-#   create incidents for reports with any failed changes or corrective changes.
-#   The `unchanged` value will create an incident for any report that does not
-#   have _any_ changes (failed, intentional, corrective, _or_ pending)0. Use
-#   this value to create an incident, even if there was Puppet found no
-#   configuration discrepencies and there were no failures. Turn turn off
-#   incident reporting completely, just specify an empty array ([]) for this
-#   parameter, and this module will not create any incidents at all.
+# @param [Servicenow_reporting_integration::IncidentCreationConditions] incident_creation_conditions
```

Comment:
Looking at the `REFERENCE.md` generated from `bundle exec puppet strings generate --format markdown`, I think just mentioning the typedef is good enough (so we avoid specifying the enums twice). The docs will also document the typedef in detail so I don't think it will be hard for users to figure out what this means.

Reply:
I also moved over the enum documentation to the typedef (`types/incidentcreationconditions.pp`).